### PR TITLE
Implementing core runtime

### LINKED ISSUE #64: implement controllers for the core runtime

### PR DESCRIPTION
**Pull request by scothis**
Restores basic controller behavior for the core runtime.

- creates Deployment and Service child resources for a Deployer
- resolves images from build references
- tracks build resources for new images, updating the deployment

Resolves #64

### CODE REVIEW

**Comment by scothis** [comment]
This PR is now functional. It will create the child Deployment and Service, resolves images from build resources and tracks the referenced resources for updated images.

Error handling is still a bit dodgy, particularly with regards to what errors get requeued.

File: pkg/apis/accessor.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	GetOwnerReferences() []metav1.OwnerReference
+	SetOwnerReferences([]metav1.OwnerReference)
+	GetClusterName() string
+	SetClusterName(clusterName string)
+}
+
+// DeletionHandlingAccessor tries to convert given interface into Accessor first;
+// and to handle deletion, it try to fetch info from DeletedFinalStateUnknown on failure.
+// The name is a reference to cache.DeletionHandlingMetaNamespaceKeyFunc
+func DeletionHandlingAccessor(obj interface{}) (Accessor, error) {
```
Comment (sbawaska) [inline pkg/apis/accessor.go]: I don't understand why this is needed.

Comment (scothis) [reply pkg/apis/accessor.go]: I'll remove it. It was a holdover from when I was attempting to use the Knative Tracker api directly.

File: pkg/controllers/core/deployer_controller.go [hunk 2]
```diff
@@ -36,18 +48,365 @@ type DeployerReconciler struct {
...
+
+	// overwrite fields that should not be mutated
+	desiredDeployment.Spec.Replicas = actualDeployment.Spec.Replicas
+
+	if r.deploymentSemanticEquals(desiredDeployment, &actualDeployment) {
+		// service is unchanged
+		return &actualDeployment, nil
+	}
+
+	// update service with desired changes
```
Comment (ericbottard) [inline pkg/controllers/core/deployer_controller.go]: should read "deployment"

**Pull request merged**